### PR TITLE
Add appveyor & error macro

### PR DESCRIPTION
**Pull request by alesiong**
Don't hasten to add the appveyor ci to the require list, as our codes fit not so well with windows

### CODE REVIEW

**Comment by Yexiaoxing** [comment]
#43 rely on it to get test passed...

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+install:
+  - git submodule update --init --recursive
+
+before_build:
+- ps: |
+    md build
+    cd build
+    $generator = switch ($env:TOOLSET)
+    {
+        "v140" {"Visual Studio 14 2015"}
```
Comment (Yexiaoxing) [inline appveyor.yml]: What about VS 2017?

Comment (alesiong) [inline appveyor.yml]: The appveyor does not support vs 2017 very well, I'm working on it.

Comment (Yexiaoxing) [inline appveyor.yml]: so let's keep it simple and not considering 2017 for not (there should be no problem)

Comment (alesiong) [inline appveyor.yml]: This modification would be big, need to be test on another testing project, and I will open another pr if it works then

**Pull request merged**